### PR TITLE
Add response types

### LINKED ISSUE #928: Swagger: add response types
Posting here as per @FahdSaif's request.

The swagger currently only shows the shapes of the requests but not the responses.

### PR DESCRIPTION
**Pull request by moisses89**
Related #928 
Add response types for: 
- collectibles v1
- about
- delegates
- safe_apps

### CODE REVIEW

File: src/routes/about/routes.rs [hunk 1]
```diff
@@ -93,15 +96,12 @@ pub async fn get_master_copies(
...
-    context: RequestContext,
-    chain_id: String,
-) -> ApiResult<content::RawJson<String>> {
+pub async fn backbone(context: RequestContext, chain_id: String) -> ApiResult<Json<String>> {
     let client = context.http_client();
     let info_provider = DefaultInfoProvider::new(chain_id.as_str(), &context);
     let url = core_uri!(info_provider, "/v1/about/")?;
     let request = Request::new(url);
-    Ok(content::RawJson(client.get(request).await?.body))
+    Ok(Json(client.get(request).await?.body))
```
Comment (moisses89) [inline src/routes/about/routes.rs:104]: This should return Json<BackBoneType>

**Comment by moisses89** [comment]
@fmrsabino regarding this fragment of code:
```
CacheResponse::new(&context, ChainCache::from(chain_id.as_str()))
        .duration(about_cache_duration())
        .resp_generator(|| handlers::get_master_copies(&context, chain_id.as_str()))
        .execute()
        .await
```
CacheResponse->execute returns `ApiResult<RawJson<String>>>` 
Instead rewrite execute I would like to add a method that returns from cache the result in String and in each endpoint cast the result to `ApiResult<Json<CorrespondingType>>>`
What do you think about this way to do this?

**Comment by fmrsabino** [comment]
> @fmrsabino regarding this fragment of code:
> 
> ```
> CacheResponse::new(&context, ChainCache::from(chain_id.as_str()))
>         .duration(about_cache_duration())
>         .resp_generator(|| handlers::get_master_copies(&context, chain_id.as_str()))
>         .execute()
>         .await
> ```
> 
> CacheResponse->execute returns `ApiResult<RawJson<String>>>` Instead rewrite execute I would like to add a method that returns from cache the result in String and in each endpoint cast the result to `ApiResult<Json<CorrespondingType>>>` What do you think about this way to do this?

Instead of casting I'd probably have it as a Generic output. This might be a bigger change however so you can also try parsing the `String` to a specific type using `serde` (altough this might result in unecessary serializations when we return from the route:  `String` -> `T` -> `U`)

**Comment by fmrsabino** [comment]
> Closes #928

This PR shouldn't close this issue. The issue is more about having it on all endpoint and this is specific to the About the Collectibles

File: src/routes/collectibles/handlers.rs [hunk 2]
```diff
@@ -26,13 +26,13 @@ pub async fn collectibles(
...
-        RequestCached::new_from_context(url, &context, ChainCache::from(chain_id))
-            .request_timeout(collectibles_request_timeout())
-            .execute()
-            .await?,
-    ))
+    let body = RequestCached::new_from_context(url, &context, ChainCache::from(chain_id))
+        .request_timeout(collectibles_request_timeout())
+        .execute()
+        .await?;
+    Ok(Json(serde_json::from_str::<Vec<ServiceCollectible>>(
```
Comment (rmeissner) [inline src/routes/collectibles/handlers.rs:33]: This doesn't make any sense. We first serialize them as a string and then convert them back to structs to convert them back to strings. 

With our current caching we cannot use the `Json` type.

**Review by rmeissner — CHANGES_REQUESTED** [review]

File: src/routes/collectibles/handlers.rs (hunk 2 above)
Comment (rmeissner) [reply src/routes/collectibles/handlers.rs:33]: Just to provider better documentation I would not add this additional parsing overhead

**Comment by fmrsabino** [comment]
We are targeting this work in https://github.com/5afe/safe-client-gateway-nest

**Pull request closed without merging**